can_tools/scrapers/official/DE/de_vaccine.py

```diff
@@ -140,3 +139,125 @@ def normalize(self, data):
+            # Parse start date
+            startDate = dt.datetime(sd[0], sd[1], sd[2])
+            # Get first dose data
+            first_dose_data = tdata['series'][0]['data']
+            idx = pd.date_range(startDate, periods=len(first_dose_data), freq='d')
+            first_dose_df = pd.DataFrame(data=first_dose_data, columns=['FirstDose'], index=idx)
+            # Get second dose data
+            second_dose_data = tdata['series'][1]['data']
+            idx = pd.date_range(startDate, periods=len(second_dose_data), freq='d')
+            second_dose_df = pd.DataFrame(data=second_dose_data, columns=['SecondDose'], index=idx)
+            df = first_dose_df.join(second_dose_df)
+            df['location_name'] = county
+            dfs.append(df)
+        return pd.concat(dfs)
+
+    def _fetch_total_administered(self):
+        dfs=[]
+        total_urls = [
+            {"county": "Kent", "url": self.kent_total_url},
+            {"county": "New Castle", "url": self.new_castle_total_url},
+            {"county": "Sussex", "url": self.sussex_total_url}
+        ]
+        for curl in total_urls:
+            county = curl['county']
+            url = curl['url']
+            r = requests.get(url)
+            soup = BeautifulSoup(r.text, features='lxml')
+            tdata = json.loads(
+                soup.find(
+                    "div",
+                    {
+                        "aria-labelledby": "chart-covid-vaccine-administrations-daily-label"
+                    }
+                )['data-charts--covid-vaccine-administrations-daily-config-value']
+            )
+            sd = tdata['startDate']
+            # Parse start date
+            startDate = dt.datetime(sd[0], sd[1], sd[2])
+            # Get first dose data
+            total_df = None
+            for srs in tdata["series"]:
+                if srs['name'] == 'Daily Count':
+                    total_data = srs['data']
+                    idx = pd.date_range(startDate, periods=len(total_data), freq='d')
+                    total_df = pd.DataFrame(data=total_data, columns=['TotalDoses'], index=idx)
+            if total_df is None:
+                raise "Couln't get county total data"
+            df = total_df
+            df['location_name'] = county
+            dfs.append(df)
+        return pd.concat(dfs)
+
+    def fetch(self):
+        totals = self._fetch_total_administered()
+        doses = self._fetch_fully_vaccinated()
+        return {'totals': totals, 'doses': doses}
+
+    def normalize(self, data):
+
+        totals = data['totals']
+        totals = totals.reset_index().rename(columns={"index": "dt"})
+        doses = data['doses']
+        doses = doses.reset_index().rename(columns={"index": "dt"})    
+        df = totals.set_index(['dt', 'location_name']).join(doses.set_index(['dt', 'location_name'])).reset_index()
+
+        df = df.fillna(0)
+        out = self._reshape_variables(df, self.variables)
+
```

Comment:
> if we're interested in collecting cumulative data we could add something like:

I think doing new data is fine because it is specified as "new" rather than cumulative. That way users can do their own cumulative aggregation if wanted.

>Would we need to add FirstDose + SecondDose in order to match the definition of total_vaccine_initiated, since we want at least one dose? Or dose using doses instead of people make this not an issue.

I think it's a non-issue...but let's have @sglyon clarify.

Reply:
That's right on the summing to make cumulative -- we'll report what the dashboard does and then deal with summations later in the pipeline (outside of the can-scrapers repo)

- - -

As for the summing of firstDose and SecondDose --- I actually don't know on this one. Ideally we want (total_vaccine_initiated, new, doses) to be pfiezer_first + moderna_first + jj_shots and (total_Vaccine_completed, new, doses) to be pfeizer_second + moderna_second + jj_shots

I don't think we are computing exactly that, are we?

Reply:
It's not entirely clear what is included in the data in the graphs [here](https://myhealthycommunity.dhss.delaware.gov/locations/state/vaccine-tracker). 

In the "Persons Fully Vaccinated in Delaware" section, the data is described as:

> The data in this section are about all vaccinations administered in Delaware. "Fully vaccinated" individuals include both residents and non-residents who received 2 doses of a 2-dose vaccine series (for example the Moderna and Pfizer vaccines) and residents and non-residents who received 1 dose of a single-dose vaccine series (for example the Johnson & Johnson vaccine).

So the raw number for _total_ number can be calculate for our definition. However, the graph to the right of the numbers, only have 1st-dose and 2nd-dose in the legend. It's unclear if 1st dose is including J&J doses or not.